docs/supported_media.md

```diff
@@ -0,0 +1,311 @@
+necessary.
+
+Publishers would need to upload their sniffing algorithms to caches, either
+through bilateral coordination, or by publishing them on the web somewhere
+(e.g. a `.well-known` URI). In some cases, this may be infeasible, e.g. if the
+publisher relies on a closed-source third-party sniffer.
+
+While it would allow for more flexibility in responsiveness than media queries,
+its complexity is in disproportion to the need.
+
+These considerations don't change with regard to the `Sec-CH-UA` proposal.
+
+#### Arbitrary media queries
+
+Referrers could allow any media queries rather than the subset specified above.
+This creates a few risks:
+
+  - Queries might evaluate incorrectly. Of the set of browsers for which a
+    referrer provides caching, the referrer would need to keep track of which
+    media types/features each supports. This complexity is disproportionate to
+    the need.
+  - Queries might hurt user experience on the referring page. For instance,
+    overly long queries may peg the CPU. (Or perhaps there are some
+    expensive-to-compute features? These features might be safe to add to the
+    web platform where a page author may choose the appropriate resource usage
+    trade-offs, but not in a shared environment containing multiple queries
+    from different authors.)
+  - Queries might vary over time. For instance, `orientation` may evaluate to
+    `portait` while on the referrer, but the user may change it to `landscape`
+    after visiting the referent. (This is true of some of the allowed subset,
+    too, but it seemed like a reasonable trade-off between utility and risk of
+    inaccuracy.)
+
+The above subset was chosen as a safe starting point, based on a cursory
+examination of common media queries. Proposals for additions are welcome.
+
+#### Predefined aliases
+
+The supported-media value could be a comma-separated list of form factors such
+as `desktop`, `mobile`, and `tablet`. Referrers and publishers would need to
+standardize such a list. This is likely difficult to do for many reasons; for
+instance, different regions may have different standards for mobile sizes. In
+addition, doing so risks ossification of the predefined aliases, which then may
+become outdated as user expectations change; for instance, phones might change
+size distribution, or users might expect specialized [foldable
+interfaces](https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/Foldables/explainer.md)
+in the future.
+
+#### meta name=viewport
+
+Referrers could use `<meta name=viewport>` as an indication that a site is
+responsive. This is an opt-in mechanism; lacking this signal, referrers would have to
+assume the site is non-responsive. See the [Assume non-responsive by
+default](#assume-non-responsive-by-default) for why this is not preferred.
+
+#### Response header
+
+Referrers could look for a `supported-media` HTTP response header rather than
```

Comment:
One could argue that an either-or approach could be used, but I think it's fair to simplify too.

Reply:
Both good points; I added a short para.